### ISSUE TITLE
In-notebook UI for creating new code/markdown cells

### ISSUE THREAD

**Issue by ellisonbg** [jupyterlab/jupyterlab#2110]
This is a more specific issue to track our design/implementation of an inline UI for making it easy to create new code/markdown cells. @cameronoelsen can you post latest designs here?

**Comment by cameronoelsen** [comment]
Just for the sake of keeping everything in the same place, here are the earlier UIs that we explored for the new Cell UI:

![f532bd88-1e99-11e7-81d0-5729d75090aa](https://cloud.githubusercontent.com/assets/6437976/25630892/7faead98-2f23-11e7-916a-20353e903a8c.gif)

![3797efb4-242f-11e7-90c5-a5d4d00e0a27](https://cloud.githubusercontent.com/assets/6437976/25630899/84bc9336-2f23-11e7-8ba3-d5698185adfd.gif)

Here is another idea for the new cell UI, we could utilize the floating action button from material design:

![ezgif-1-4e5733d732](https://cloud.githubusercontent.com/assets/6437976/25630926/9c3de802-2f23-11e7-9587-e24ffd7b0bbc.gif)

While I do think the color added by the material design floating action buttons nicely contrasts with the whiteness of the document, they feel out of place inside the notebook UI

**Comment by scottdraves** [comment]
Something like this would be an awesome improvement of the notebook UI.

Of these 3 I like the one in the middle the most.
In #1 Moving the cells on hover seems disruptive.
In #3 Having 2 stages is too slow.

So I Like the style of #2 the best however I would make the representations of the two cell types (code and markdown) clearer and simpler.  I don't believe `</>` is universally understood to be code, especially to a Python programmer.  Also, it maybe a application-wide change but "md+" is really technical and probably not what most users think of.  Maybe just "text"?

**Comment by ellisonbg** [comment]
Yeah, we are continuing to iterate on the design of this. Some good ideas here, but haven't quite found the right thing. Fully agree with your sense @scottdraves

**Comment by ellisonbg** [comment]
Marking this as beta - we might have time to do it and it is high value. Can be deferred if needed.

**Comment by JasonWeill** [comment]
The cell toolbar (#12028) added "create new cell above" and "create new cell below" features. @HaudinFlorence 's work in #14109 added a new-cell button at the bottom of a workbook. I can keep this issue open to explore further enhancements.
